api/v1alpha1/stage_helpers.go

```diff
@@ -136,39 +136,28 @@ func ReverifyStageFreight(
 		return err
 	}
 
-	current := stage.Status.FreightHistory.Current()
-	if current == nil || len(current.Freight) == 0 {
+	currentFC := stage.Status.FreightHistory.Current()
+	if currentFC == nil || len(currentFC.Freight) == 0 {
 		return errors.New("stage has no current freight")
 	}
 
-	if len(current.Freight) > 1 {
-		// TODO: This is a naive implementation which does not handle multiple
-		// Freight objects in the history properly. This must be addressed for
-		// proper "multi-pipeline" support.
-		return errors.New("stage has multiple Freight objects: re-verification not supported at this time")
+	currentVI := currentFC.VerificationHistory.Current()
+	if currentVI == nil {
+		return errors.New("stage has no current verification info")
 	}
 
-	for _, f := range current.Freight {
```

Comment:
This loop pre-dated the VerificationHistory moving up a level from Freight to FreightCollection. The change made for aborting non-terminal verifications is basically the same loop removal.